record_api/infer_apis.py

```diff
@@ -42,7 +42,7 @@ def __main__():
 
 
 def parse_line(
-    n: int, function: object, params=None, bound_params=None,
+    n: int, function: object, params=None, bound_params=None, return_type=None
```

Comment:
This is fine for now. We will have to add something to the signature to represent the return type and then we should display it as a return type annotation, see [PEP 484](https://www.python.org/dev/peps/pep-0484/#abstract):

```python
def greeting(name: str) -> str:
    return 'Hello ' + name
```

Reply:
It can be anything at the moment, as we don't have a specialised object. Do you mean to say adding `Any` hint?

Reply:
Yeah for most opcodes we don't k ow the return type yet, but as you implement them, we will...

The whole "how do we type things and then combine them" is another part of the code, totally separate from the recording.

We can use the [`ObjectOutput`](https://github.com/data-apis/python-record-api/blob/4fa06df6c0eac85e9f54d70f7172dde9e1ef3125/record_api/type_analysis.py#L343-L352) for places we don't know the return type yet. It is the same as `typing.Any`.